pkg/api/endpoints/util.go

```diff
@@ -28,6 +28,18 @@ import (
 	hashutil "k8s.io/kubernetes/pkg/util/hash"
 )
 
+const (
+	PodHostnameAnnotation  = "net.beta.kubernetes.io/podHostname"
+	PodSubdomainAnnotation = "net.beta.kubernetes.io/podSubdomain"
+	PodHostNamesAnnotation = "net.beta.kubernetes.io/podHostnames"
+)
+
+type HostFQDN struct {
+	HostName    string
+	Namespace   string
```

Comment:
remind me why we need namespace and service name here?  The controller should have that from context?